src/js/common/schemaform/reducers/index.js

```diff
@@ -21,12 +21,19 @@ function recalculateSchemaAndData(initialState) {
     .reduce((state, pageKey) => {
       const page = state[pageKey];
       // on each data change, we need to do the following steps
+
+      // Flatten the data from all the pages
+      const formData = _.mapValues('data', state);
```

Comment:
I tried this branch out on my Moto G4 and I don't see any noticeable slow down compared to my testing on our other forms. I didn't really expect this one change to be noticeable, but I hadn't tried out either of the new forms, which are bigger than the 1995, at all yet. 

As far as encouraging dependencies between pages, probably doesn't make sense to make it more awkward for the developer when how the validation works is more of a UX decision. 

I think we should just be cautious when building forms if we start to add lots of complicated dependencies and conditional logic.